students/rhamersky/Lesson_6/mailroom_part4.py

```diff
@@ -0,0 +1,263 @@
+        while True:
+            # Appends the donor list.
+            if donor_name == "List":
+                # --> Verifies that there is at least a donor is stored as a list row in the table tuple.
+                if len(dic_donors) > 0:
+                    print_donor_names()  # --> Prints a list of donors
+                    break
+                else:
+                    print(
+                        "\n" + "There are no donors in list.")  # --> Let's the user know that a donor list does not exist.
+                    break
+
+            try:
+                flo_donation_amount = float(input(str_name + ", please enter donation amount. "))
+
+                # Updates donor tuple
+                dic_donors = update_donor(donor_name, flo_donation_amount)
+
+                # String formatting for a brief thank-you to donor.
+                thankyou_message = "\n"+"{}, thank you for donating ${:,.2f}.".format(donor_name, flo_donation_amount)
+                print(thankyou_message)
+                break
+
+            except ValueError:  # --> Allows the program to continue if the user does not enter the right data type.
+                print(str_name + ", not a valid amount. \n")
+                continue
+
+    @staticmethod
+    def create_report():
+        '''
+        This method will create a report for the user with the following columns:
+        donor's name, number of gifts, average gift amount, and lifetime donations.
+        :return:
+        '''
+        longest_entry = float()
+        # Finds the max amount of characters for all items stored in dic_donors.
+        for name in dic_donors:
+            entry_length = len(name)
+            if longest_entry < entry_length:
+                longest_entry = entry_length
+            for index in range(0,1):
+                entry_length = len(str(dic_donors[name][index]))
+                if longest_entry < entry_length:
+                    longest_entry = entry_length
+
+        lst_headers = ["Donor_Name", "Num_Gifts", "Average_Gift", "Total_Given"]  # --> List of headers for the chart.
+
+        # Creates the header for the chart
+        for heading in lst_headers:
+            print(("{0:{padding}}".format(heading, padding=longest_entry)), end=" ")
+        print()
+
+        # Creates the donor rows for the chart.
+        sorted_dic = dic_sort()
+        for name in sorted_dic:
+            donation_avg = dic_donors[name][0] / dic_donors[name][1]
+            print(("{0:{padding}}".format(name, padding=longest_entry)), end=" ")
+            print(("{0:{padding}}".format("{}".format(dic_donors[name][1]), padding=longest_entry)), end=" ")
+            print(("{0:{padding}}".format("${:,.2f}".format(donation_avg), padding=longest_entry)), end=" ")
+            print(("{0:{padding}}".format("${:,.2f}".format(dic_donors[name][0]), padding=longest_entry)), end=" ")
+            print()
```

Comment:
You could build up a string to return, rather than printing line by line to test the content of the report